File: self-check.py
# -*- coding: utf-8 -*-
# 보안을 위해 일부 정보가 삭제되었습니다.
# Syun 제작

import json,re,urllib.request as u
from urllib import parse as p
q=p.quote

def lambda_handler(event,context):
    params=json.loads(event['body'])['action']['params']
    name=q(params['name'])
    pw=q(params['pw'])
    birth=q(json.loads(params['birth'])['value'].replace('-','')[2:])
    
    with u.urlopen('주소') as url:
        r=url.read().decode()
    
    try:
        json.loads(r)['registerDtm']
        d='자가진단이 완료되었습니다.'
    except:
        d='자가진단이 실패하였습니다.'
    
    result={
        'version':'2.0',
        'data':{
            'd':d
        }
    }

    return {
        'statusCode':200,
        'body':json.dumps(result),
        'headers':{'Access-Control-Allow-Origin':'*'}}

File: selfcheck.php
// 제작: Syun
// 본 소스를 이용해 생긴 일에 대한 책임은 사용자에게 있습니다.
// 본 소스에는 제작자의 개인정보 보호를 위해 일부 정보가 제거되었습니다.

<?php
include 'key.php';

$params=json_decode(file_get_contents('php://input'))->action->params;
$name = $params->name;
$birth=substr(str_replace('-' , '', json_decode($params->birth)->value),2);;
$pw = $params->pw;

$url = file_get_contents('**** URL ****');

if (isset(json_decode($url)->registerDtm)) {
    $success='성공';
} else {
    $success='실패';
}

$result = [
	'version' => '2.0', 
	'data' => [
		'd'=>$success
	]
];

echo json_encode($result,JSON_UNESCAPED_UNICODE);
?>
